playbooks/roles/genesis-exoscale/tasks/main.yml

```diff
@@ -0,0 +1,178 @@
+---
+- name: Set the Exoscale API details facts to the value that was entered, or attempt to retrieve it from the environment if the entry is blank
+  set_fact:
+    EXO_API_SECRET: "{{ EXO_API_SECRET_entry | default( lookup('env', 'EXO_API_SECRET') ) }}"
+    EXO_API_KEY: "{{ EXO_API_KEY_entry | default( lookup('env', 'EXO_API_KEY') ) }}"
+    EXO_API_ENDPOINT: "{{ lookup('env', 'EXO_API_ENDPOINT') | default('https://api.exoscale.ch/compute', true) }}"
+
+
+- name: Get the default SSH key
+  command: cat ~/.ssh/id_rsa.pub
+  register: ssh_key
+
+- block:
+    - name: Add the SSH key to Exoscale if it doesn't already exist
+      local_action:
```

Comment:
Because these tasks are executing within the context of a local playbook, the `local_action` syntax is not strictly necessary. We should be able to do something like this:

```yaml
  - name: Create the Streisand Security Group
    cs_securitygroup:
      name: "streisand"
      description: "Streisand VPN"
```